src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,59 @@
+package duke;
+
+import java.io.IOException;
+public class Parser {
+    /**
+     *
+     * @param str description of the task
+     * @param taskList the taskList where tasks are stored and loaded from
+     * @param ui the interactive UI system
+     * @param storage storage deals with the writing and loading of files
+     * @return boolean
+     * @throws DukeException
+     */
+    public static boolean isParse(String str, TaskList taskList, Ui ui, Storage storage) throws DukeException {
+
+            try {
+                if (str.equals("bye")) {
+                    ui.bye();
+                    return true;
+                }
+                else if (str.equals("list")) {
```

Comment:
Aside from comments by other reviewers, perhaps you should consider putting your closing curly brace for the `if` statement on the same line as `else {` to conform to the [coding guidelines](https://se-education.org/guides/conventions/java/intermediate.html#layout)?